### PR TITLE
[3.x]Upgrade Slurm to 22.05.3

### PR DESCRIPTION
**Pull request by hanwen-pcluste**
Signed-off-by: Hanwen <hanwenli@amazon.com>

### Tests
Integration tests in `configs/pcluster3.yaml` and `test_cluster_in_no_internet_subnet` have been passed

### References
https://github.com/aws/aws-parallelcluster-cookbook/pull/1527 should be merged together with this PR

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration-tests/tests/networking/test_cluster_networking.py [hunk 1]
```diff
@@ -136,16 +136,7 @@ def test_cluster_in_no_internet_subnet(
...
-        "HeadNode": {"cfn-init", "cloud-init", "clustermgtd", "chef-client", "slurmctld", "supervisord"},
-        "Compute": {
-            "syslog" if os.startswith("ubuntu") else "system-messages",
-            "computemgtd",
-            "supervisord",
-            "slurm_prolog_epilog",
-        },
-    }
-    utils.check_pcluster_list_cluster_log_streams(cluster, os, expected_log_streams)
+    utils.check_pcluster_list_cluster_log_streams(cluster, os)
```
Comment (jdeamicis) [inline tests/integration-tests/tests/networking/test_cluster_networking.py:139]: I assume this is due to a rebase, right? If not, could we please put this in a separate commit?

**Review by jdeamicis — DISMISSED** [review]

File: tests/integration-tests/tests/networking/test_cluster_networking.py (hunk 1 above)
Comment (hanwen-pcluste) [reply tests/integration-tests/tests/networking/test_cluster_networking.py:139]: This is not because of a rebase. We are removing the workaround for a Slurm bug in https://github.com/aws/aws-parallelcluster-cookbook/pull/1527. Therefore, we do not need to check `slurm_prolog_epilog` log. All other logs are in the default `expected_log_streams` in `utils.check_pcluster_list_cluster_log_streams`

**Review by jdeamicis — APPROVED** [review]
